setup/functions.sh

```diff
@@ -1162,3 +1162,40 @@ function backup_work_dir {
   fi
 }
 export -f backup_work_dir
+
+# Check if a Hugging Face model is gated (requires manual approval)
+# Usage: is_hf_model_gated <model_id>
+# Example: is_hf_model_gated ibm-granite granite-3.1-8b-instruct
+function is_hf_model_gated {
+    local model_id="$1"
+    local url="https://huggingface.co/api/models/${model_id}"
+
+    local response=$(curl -s -H "Accept: application/json" "${url}")
+    if [[ $? -ne 0 || -z "${response}" ]]; then
+        return 2
+    fi
+    
+    local gated=$(echo "${response}" | jq -r '.gated // false')
+    if [[ ${gated} == "false" ]]; then
+      return 1
+    else
+      return 0
+    fi
```

Comment:
Standard bash pattern is returning 0 for success and 1 for error. Minor nit and not required, but wondering if you want to return 0 for non-gated models and 1 for gated or maybe just add a comment for this function that 1 means non-gated?